wordpress-mobile/WordPress-Android
# [Site Creation] Align screen titles  to left (right for RTL)

Fixes #16551

Aligns the tile & subtitle texts of the new & updated site creation screens (Site topic, Site name) to the left or right in case of RTL languages.

The title on the Site theme (aka 'Choose a theme') screen was already aligned to the left/right in PR #16428.

**⚠️ Note** _This solution is one possible way to move forward at this step with regards to the new specs for text alignment. Another option would be to restore the text alignment to center for the Site Theme screen and postpone this implementation after the work on the Site Theme screen is done, since initially it was specified as being optional work for the Site Design Screen Improvement project (see internal ref pbArwn-4zp-p2)._

I also refactored the `Choose a theme` screen to include the same header layout component we use for the `Site Topic` & `Site Name` screens.

<details>
<summary>Preview</summary>

https://user-images.githubusercontent.com/4588074/168549946-c5e960b5-f643-4a07-9da4-e03ec4e411af.mp4
</details>

To test:
<details>
<summary>Prerequisite</summary>

Manually override the `isEnabled` function of `SiteNameFeatureConfig` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/1010c25bcfeac5854d0fcf0a5b89f78050789148/WordPress/src/main/java/org/wordpress/android/util/config/SiteNameFeatureConfig.kt#L19) to return `true`.
</details>

1. Start the site creation flow
   - **Expect** to land on Site Topic screen
   - **Verify** title & subtitle in header section are aligned to the left (right on RTL languages)
2. Tap the input
   - **Verify** collapsed title is aligned to the left (right on RTL languages)
3. Skip or continue with a site topic
   - **Expect** to land on Site Name screen
   - **Verify** title & subtitle in header section are aligned to the left (right on RTL languages)
4. Rotate screen to landscape orientation
   - **Verify** collapsed title is aligned to the left (right on RTL languages)
5. Skip or continue with a site name
   - **Expect** to land on Choose a theme screen
   - **Verify** collapsed title is aligned to the left (right on RTL languages)
5. Rotate screen to portrait orientation
   - **Verify** title in header section is aligned to the left (right on RTL languages)

## Regression Notes
1. Potential unintended areas of impact
   N/a

7. What I did to test those areas of impact (or what existing automated tests I relied on)
   - Manual testing of different device orientation (landscape/portrait), device sizes (phone/tablet) and language text direction (LTR/RTL) to make sure it works as described in the test steps.
   - Manual testing of the `Choose a theme` screen after refactoring to use the same header layout component we use for the Site Topic and Site Name screens.
8. What automated tests I added (or what prevented me from doing so)
   N/a - We don't usually test UI changes without business logic.

PR submission checklist:
